src/components/PostPageDetail/index.js

```diff
@@ -676,7 +676,7 @@ const PostPageDetailIdComponent = (props) => {
                 showScoreButton={profile?.myProfile?.is_backdoor_user}
                 onPressScore={handleOnPressScore}
                 onPressBlock={() => refBlockComponent.current.openBlockComponent(item)}
-                isSelf={profile.myProfile.user_id === item.actor?.id}
+                isSelf={item?.is_self}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The change from directly comparing user IDs to using the `is_self` flag is a good one. It simplifies the code and makes it more readable. However, please ensure that the `is_self` flag is reliably set in all cases where a post belongs to the current user.
```diff
- isSelf={profile.myProfile.user_id === item.actor?.id}
+ isSelf={item?.is_self}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->